npm_and_yarn/helpers/lib/npm/vulnerability-auditor.js

```diff
@@ -73,7 +73,8 @@ async function findVulnerableDependencies(directory, advisories) {
       return response
     }
     const vuln = auditReport.get(name)
-    const version = [...vuln.nodes][0].version
+    const vulnNodesHighestLevelFirst = [...vuln.nodes].sort((a, b) => a.depth - b.depth);
```

Comment:
Not sure if it's even possible, but what happens here with a race condition? Where 2 versions of the dependency are the same depth in the tree?

Reply:
That shouldn't be possible because there can only be one version installed at root level. There can certainly be multiple copies installed at deeper levels if they are from different branches of a dependency tree, though.

But in this case, there _should_ always be a single version present at the root level for the package. I chose to make this line general just in case there is a situation in which a dependency isn't present at the root level but it is further down (but really that shouldn't ever happen). In such a case picking an arbitrary copy of the dependency is the best we can do anyway.